src/sentry/audit_log/manager.py

```diff
@@ -0,0 +1,89 @@
+from dataclasses import dataclass
+from typing import Callable, List, Optional
+
+import sentry_sdk
+
+from sentry.models.auditlogentry import AuditLogEntry
+
+
+@dataclass
+class AuditLogEvent:
+    event_id: int
+    name: str
+    api_name: str
+
+    # Use `template` OR `get_template` for each AuditLogEvent.
+
+    # For audit log events using one string format to display the log's message,
+    # use `temaplate`.
+    # Add any new variables to `template_variables` in `AuditLogEventManager.get_note`.
+    template: Optional[str] = None
+
+    # For audit log events that need a function to determine the log's message,
+    # use `get_template` to store that function.
+    get_template: Optional[Callable[["AuditLogEntry"], str]] = None
+
+
+class AuditLogEventManager:
+    _event_registry = {}
+    _event_id_lookup = {}
+
+    def add(self, audit_log_event: "AuditLogEvent"):
```

Comment:
```suggestion
    def add(self, audit_log_event: AuditLogEvent):
```

You should be able to use the class object as the typehint here.